tests/resourcecollection_test.go

```diff
@@ -0,0 +1,499 @@
+		{
+			"namespace with embedded",
+			&v1.DryRunCollectionRequest{
+				Name: "test collection",
+				ResourceSelectors: []*storage.ResourceSelector{
+					{
+						Rules: []*storage.SelectorRule{
+							{
+								FieldName: search.Namespace.String(),
+								Operator:  storage.BooleanOperator_OR,
+								Values: []*storage.RuleValue{
+									{
+										Value: collectionNamespaces[0],
+									},
+								},
+							},
+						},
+					},
+				},
+				EmbeddedCollectionIds: []string{collectionIDs[0]},
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment1" || deployment.GetNamespace() == collectionNamespaces[0]
+			}),
+		},
+		{
+			"namespace with deployment name",
+			&v1.DryRunCollectionRequest{
+				Name: "test collection",
+				ResourceSelectors: []*storage.ResourceSelector{
+					{
+						Rules: []*storage.SelectorRule{
+							{
+								FieldName: search.Namespace.String(),
+								Operator:  storage.BooleanOperator_OR,
+								Values: []*storage.RuleValue{
+									{
+										Value: collectionNamespaces[0],
+									},
+								},
+							},
+							{
+								FieldName: search.DeploymentName.String(),
+								Operator:  storage.BooleanOperator_OR,
+								Values: []*storage.RuleValue{
+									{
+										Value: "deployment2",
+									},
+								},
+							},
+						},
+					},
+				},
+				EmbeddedCollectionIds: []string{collectionIDs[0]},
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment1" || (deployment.GetNamespace() == collectionNamespaces[0] && deployment.GetName() == "deployment2")
+			}),
+		},
+		{
+			"embedded collection",
+			&v1.DryRunCollectionRequest{
+				Name:                  "test collection",
+				EmbeddedCollectionIds: []string{collectionIDs[1]},
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment2"
+			}),
+		},
+		{
+			"embedded collections",
+			&v1.DryRunCollectionRequest{
+				Name:                  "test collection",
+				EmbeddedCollectionIds: collectionIDs,
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment1" || deployment.GetName() == "deployment2"
+			}),
+		},
+	} {
+		s.T().Run(tc.name, func(t *testing.T) {
+			resp, err := s.service.DryRunCollection(ctx, tc.request)
+			assert.NoError(t, err)
+			assert.ElementsMatch(t, tc.expectedListDeployments, resp.GetDeployments())
+		})
+	}
+
+	// clean up test
+	for _, id := range collectionIDs {
```

Comment:
I think this block may not run if test fails in between collection creation? What about putting it in `TearDownTest` (and the collection creation to `SetupTest`)?

Or put this in a defer block

Reply:
Moved the embedded collection creation to setup and the teardown now includes cleaning those up